htdocs/core/lib/security.lib.php

```diff
@@ -1051,7 +1051,7 @@ function checkUserAccessToObject($user, array $featuresarray, $object = 0, $tabl
 				$sql .= " FROM ".MAIN_DB_PREFIX.$dbtablename." as dbt";
 				$sql .= " WHERE dbt.rowid IN (".$db->sanitize($objectid, 1).")";
 				$sql .= " AND dbt.".$dbt_keyfield." = ".((int) $user->socid);
-			} elseif (isModEnabled("societe") && empty($user->rights->societe->client->voir)) {
+			} elseif (isModEnabled("societe") && empty($user->rights->societe->client->voir)&& ($feature != 'knowledgemanagement')) {
```

Comment:
I think a better fix would be to add
'knowledgemanagement' into the array $check line 895. Can you test this ?

Reply:
It works

Reply:
Thanks for this proposal. It works and I updated the PR

Reply:
@eldy May I ask you to merge this PR or tell me what I can do to get it merged